Automattic/wp-calypso
# Contact form: Limit the blocks that can be inserted to only core blocks and markdown

This make sure that the user is only able to use some of the blocks. 

#### Changes proposed in this Pull Request
Adds a whitelist of blocks that a user can add to the form. 
This prevents cases where the user enter blocks that are currently broken and lead to a broken user flow. 

For example a user can right now insets a form into a form. 

Check the code for the list of supported blocks. 
I removed, button, columns, and the contact form. 

#### Testing instructions
Testing instructions

1. Check out this Jurassic Ninja link.
    https://jurassic.ninja/create/?gutenberg&gutenpack&shortlived&jetpack-beta&calypsobranch=add/contact-form-block-limit-available-blocks&branch=add/contact-form-gutenblock-sdk

2. Connect jetpack

3. Start a new page

4. Add a the new contact form block

5. Try adding all the fields. Can you see one that breaks everything on the frontend? That doesn't make sense?
